x-pack/plugins/apm/server/lib/connections/get_connection_metrics/get_destination_map.ts

```diff
@@ -0,0 +1,228 @@
+      const sample = bucket.sample.top[0].metrics;
+
+      const spanId = sample[SPAN_ID] as string;
+
+      destinationsBySpanId.set(spanId, {
+        backendName: bucket.key.backendName as string,
+        spanId,
+        spanType: (sample[SPAN_TYPE] as string | null) || '',
+        spanSubtype: (sample[SPAN_SUBTYPE] as string | null) || '',
+      });
+    });
+
+    const transactionResponse = await apmEventClient.search(
+      'get_transactions_for_exit_spans',
+      {
+        apm: {
+          events: [ProcessorEvent.transaction],
+        },
+        body: {
+          query: {
+            bool: {
+              filter: [
+                {
+                  terms: {
+                    [PARENT_ID]: Array.from(destinationsBySpanId.keys()),
+                  },
+                },
+                // add a 5m buffer at the end of the time range for long running spans
+                ...rangeQuery(
+                  startWithOffset,
+                  endWithOffset + 1000 * 1000 * 60 * 5
+                ),
+              ],
+            },
+          },
+          size: destinationsBySpanId.size,
+          fields: asMutableArray([
+            SERVICE_NAME,
+            SERVICE_ENVIRONMENT,
+            AGENT_NAME,
+            PARENT_ID,
+          ] as const),
+          _source: false,
+        },
+      }
+    );
+
+    transactionResponse.hits.hits.forEach((hit) => {
+      const spanId = String(hit.fields[PARENT_ID]![0]);
+      const destination = destinationsBySpanId.get(spanId);
+
+      if (destination) {
+        destinationsBySpanId.set(spanId, {
+          ...destination,
+          serviceName: String(hit.fields[SERVICE_NAME]![0]),
+          environment: String(
+            hit.fields[SERVICE_ENVIRONMENT]?.[0] ??
+              ENVIRONMENT_NOT_DEFINED.value
+          ),
+          agentName: hit.fields[AGENT_NAME]![0] as AgentName,
+        });
+      }
+    });
+
+    const nodesByBackendName = new Map<string, Node>();
+
+    destinationsBySpanId.forEach((destination) => {
+      const existingDestination =
+        nodesByBackendName.get(destination.backendName) ?? {};
+
+      const mergedDestination = {
+        ...existingDestination,
+        ...destination,
+      };
+
+      let node: Node;
+      if ('serviceName' in mergedDestination) {
+        node = {
+          serviceName: mergedDestination.serviceName,
+          agentName: mergedDestination.agentName,
+          environment: mergedDestination.environment,
+          id: objectHash({ serviceName: mergedDestination.serviceName }),
+          type: NodeType.service,
+        };
+      } else {
+        node = {
+          backendName: mergedDestination.backendName,
+          spanType: mergedDestination.spanType,
+          spanSubtype: mergedDestination.spanSubtype,
+          id: objectHash({ backendName: mergedDestination.backendName }),
+          type: NodeType.backend,
+        };
+      }
```

Comment:
Can't you make node a const and use a ternary to set its value?

```suggestion
      const node: Node = 'serviceName' in mergedDestination ? {
          serviceName: mergedDestination.serviceName,
          agentName: mergedDestination.agentName,
          environment: mergedDestination.environment,
          id: objectHash({ serviceName: mergedDestination.serviceName }),
          type: NodeType.service,
        } : 
        {
          backendName: mergedDestination.backendName,
          spanType: mergedDestination.spanType,
          spanSubtype: mergedDestination.spanSubtype,
          id: objectHash({ backendName: mergedDestination.backendName }),
          type: NodeType.backend,
        }
```

Reply:
I think the if/else is a little more readable here, due to the size of the object being created.